lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1072,27 +1077,29 @@ private static function get_setting_nodes( $theme_json, $selectors = array() ) {
 	 * Returns the stylesheet that results of processing
 	 * the theme.json structure this object represents.
 	 *
-	 * @param string $type Type of stylesheet. It accepts:
-	 *                     'all': css variables, block classes, preset classes. The default.
-	 *                     'block_styles': only block & preset classes.
-	 *                     'css_variables': only css variables.
-	 *                     'presets': only css variables and preset classes.
+	 * @param string $type   Type of stylesheet. It accepts:
+	 *                         'all': css variables, block classes, preset classes. The default.
+	 *                         'block_styles': only block & preset classes.
+	 *                         'css_variables': only css variables.
+	 *                         'presets': only css variables and preset classes.
+	 * @param array  $origins A list of origins to include. By default it includes 'core', 'theme', and 'user'.
+	 *
 	 * @return string Stylesheet.
 	 */
-	public function get_stylesheet( $type = 'all' ) {
+	public function get_stylesheet( $type = 'all', $origins = self::VALID_ORIGINS ) {
```

Comment:
Are we adding more complexity to the caller by adding new options here instead of absorbing the logic inside the class? Maybe this is not the right level of abstraction I'm looking for. When I want the stylesheet of a theme, why should I need to specify a "type" (meaning do we have dedicated use case to retrieve two different stylesheets for the same theme) or "origins"

Reply:
> meaning do we have a dedicated use case to retrieve two different stylesheets for the same theme

Yeah, we have. The front enqueues a single stylesheet with all styles but the block editor needs two different stylesheets: one with the block styles that will be wrapped by `.editor-styes-wrapper` and the other with the CSS Custom Properties that won't be wrapped so they're available to the UI controls in the sidebar.

Reply:
I'm troubled about the `origins` as well. However, this is the result of having added them to the existing format for each preset. We still have use cases when we only want one preset (stylesheet for themes without theme.json support, settings from the theme, etc).

I'll continue with the explorations at https://github.com/WordPress/gutenberg/pull/34843 to create a higher-level API for consumers.